docs/1-getting-started/6-building-a-nav-bar/README.md

```diff
@@ -0,0 +1,161 @@
+
+![Nested pages](nested-pages.png)
+
+Our goal now is to build a navigation bar that displays the first two levels of the page tree. The navigation bar should be dynamic and update automatically when the page tree changes.
+
+We'll create a new component called `navBar` with a simple definition. Create the file `src/components/NavBar/defintion.cnd` with the following content:
+
+```cnd
+[hydrogen:navBar] > hydrogenmix:component, jnt:content
+```
+
+It doesn't have any properties, but you can add some if you want to customize the navigation bar further.
+
+Our `NavBar` component will create the following HTML structure:
+
+```html
+<nav>
+  <ul>
+    <li>
+      <a href="/home">Home</a>
+      <ul>
+        <li>
+          <a href="/home/new-products">New Products</a>
+        </li>
+        <li>
+          <a href="/home/case-studies">Case Studies</a>
+        </li>
+      </ul>
+    </li>
+    ...
+  </ul>
+</nav>
+```
+
+To get all these pages, we'll start at the site level, `renderContext.getSite()`, and iterate over all the child pages. We'll do it a second time to get the child pages of the child pages.
+
+Iterating over children can be done with the `getChildNodes` function. It takes a node and a filtering function as arguments. The filtering function should return `true` if the node should be included in the result. We'll only keep nodes of type `jnt:page`.
+
+Create the file `src/components/NavBar/NavBar.server.tsx` with the following content:
+
+```tsx
+import { buildNodeUrl, getChildNodes, jahiaComponent } from "@jahia/javascript-modules-library";
+import type { JCRNodeWrapper } from "org.jahia.services.content";
+
+/** Get all child pages of a node. */
+const getChildPages = (node: JCRNodeWrapper) =>
+  getChildNodes(node, -1, 0, (node) => node.isNodeType("jnt:page"));
+
+jahiaComponent(
+  {
+    componentType: "view",
+    nodeType: "hydrogen:navBar",
+    displayName: "NavBar",
+  },
+  (_, { renderContext, mainNode }) => (
+    <nav>
+      <ul>
+        {getChildPages(renderContext.getSite().getNode()).map((page) => (
+          <li key={page.getPath()}>
+            <a href={buildNodeUrl(page)} aria-current={page === mainNode ? "page" : undefined}>
+              {page.getProperty("jcr:title").getString()}
+            </a>
+            <ul>
+              {getChildPages(page).map((page) => (
+                <li key={page.getPath()}>
+                  <a
+                    href={buildNodeUrl(page)}
+                    aria-current={page === mainNode ? "page" : undefined}
+                  >
+                    {page.getProperty("jcr:title").getString()}
+                  </a>
+                </li>
+              ))}
+            </ul>
+          </li>
+        ))}
+      </ul>
+    </nav>
+  ),
+);
+```
+
+We use `aria-current="page"` rather than a CSS class to indicate the current page to make the navigation bar accessible.
```

Comment:
```suggestion
We use [`aria-current="page"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-current#page) rather than a CSS class to indicate the current page to make the navigation bar accessible.
```